lib/data/external/AwsClient.js

```diff
@@ -64,6 +52,23 @@ class AwsClient {
             return callback(null, awsKey, dataStoreVersionId);
         };
 
+        const params = {
+            Bucket: this._awsBucketName,
+            Key: awsKey,
+        };
+        if (keyContext.isDeleteMarker) {
```

Comment:
it's a put of a delete marker?

Reply:
Yes 👍

Reply:
please just add a comment stating that we call put to create a delete marker but in calling AWS it's actually a delete to them